<!-- bring up perception nodes for rhino log or real time test --> 

<launch>
    <!-- args -->
    <arg name="robot_name" default="rhino" />
    <arg name="urdf_model" default="$(find common)/config/vehicles/$(arg robot_name)/urdf/car.urdf" />
    <arg name="system_setup" default="rhino_real" />
    <arg name="track_name" default="rural_test_route_1" />

    <!-- params -->
    <param name="robot_name" type="string" value="$(arg robot_name)" />
    <param name="system_setup" type="string" value="$(arg system_setup)" />
    <param name="track_name" type="string" value="$(arg track_name)" />	
    <rosparam command="load" file="$(find common)/config/vehicles/$(arg robot_name)/config/car.yaml" />
    <rosparam command="load" file="$(find perception)/config/perception_config.yaml" />
    
    <!-- perception nodes -->
    <node pkg="perception" type="track_iface.py" name="track_iface" output="screen" > </node>
    <node pkg="perception" type="state_est_cart.py" name="state_est_cart" output="screen"> </node>
    <node pkg="perception" type="state_est_frenet.py" name="state_est_frenet" output="screen"> </node>
    <node pkg="perception" type="road_perception.py" name="road_perception" > </node>

    <!-- visualization -->
    <param name="robot_description_simple" command="$(find xacro)/xacro --inorder $(arg urdf_model)" />
    <node name="rviz_" pkg="rviz" type="rviz" args="-d $(find common)/config/rviz_config/saarti_real.rviz "/>

</launch>
